gateway/engine/es/src/main/java/io/apiman/gateway/engine/es/ESMetrics.java

```diff
@@ -96,13 +110,22 @@ public void run() {
      */
     protected void processQueue() {
         try {
-            RequestMetric metric = queue.take();
-            Index index = new Index.Builder(metric).refresh(false)
-                    .index(getIndexName())
-                    .type("request").build(); //$NON-NLS-1$
-            DocumentResult result = getClient().execute(index);
+            Collection<RequestMetric> batch = new ArrayList<>(this.batchSize);
+            RequestMetric rm = queue.take();
+            batch.add(rm);
+			queue.drainTo(batch, this.batchSize - 1);
+			
+			Builder builder = new Bulk.Builder();
```

Comment:
The call to `take()` blocks until something is ready.  Then the call to `drainTo()` returns AT MOST `this.batchSize - 1` items.  But it doesn't block.  So if nothing is in the queue, then it returns immediately.  So the thread is NEVER blocked waiting for the queue to fill up to a minimum size.